src/transformers/data/data_tf_collator.py

```diff
@@ -0,0 +1,218 @@
+# Copyright 2020 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List, Optional, Tuple, Union
+
+import tensorflow as tf
+
+from ..tokenization_utils_base import BatchEncoding, PreTrainedTokenizerBase
+
+
+class TFDataCollatorForLanguageModeling:
+    """
+    Data collator used for language modeling. Encodes sequences for Masked Language Modeling as mentioned in the paper
+    'BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding'.
+
+    Labels are -100 for non-masked tokens and the value to predict the masked token.
+
+    Args:
+        tokenizer (:class:`~transformers.PreTrainedTokenizer` or :class:`~transformers.PreTrainedTokenizerFast`):
+            The tokenizer used for encoding the data.
+        batch_size (:obj:`int`):
+            The size of the batch.
+        padding_length (:obj:`int`, `optional`):
+            The length of the vector concatenated to the largest element.
+        mlm_probability (:obj:`float`, `optional`, defaults to 0.15):
+            The probability with which to (randomly) mask tokens in the input.
+        special_tokens_mask (:obj:`tf.Tensor`, `optional`):
+            If set, special tokens will have zero probability of being masked.
+    """
+
+    def __init__(
+        self,
+        tokenizer: PreTrainedTokenizerBase,
+        batch_size: int,
+        special_tokens_mask: tf.Tensor = None,
+        padding_length: int = None,
+    ):
+
+        self.tokenizer = tokenizer
+        self.batch_size = batch_size
+        self.padding_length = padding_length
+        self.special_tokens_mask = special_tokens_mask
+        self.mlm_probability = 0.15
+
+    def __post_init__(self):
+        if self.tokenizer.mask_token is None:
```

Comment:
I can see this collator only supports MLM rather than CLM, so we might need to either include CLM support, or rename it to something like `TFDataCollatorForMaskedLanguageModeling` and have a separate CLM collator. My preference would be for the former, but I realize that might cause some code complexity.

Reply:
This is something that we've discussed ourselves as well. We wanted to submit what we had to assure that you and the team agreed with our approach, and had planned on finishing out the TFDataCollator suite so it is on par with the Pytorch Data Collator. If CLM collator is highest on priority, that can be the next Data Collator @sdwalker62 and I tackle.